unstructured_ingest/v2/processes/connectors/redisdb.py

```diff
@@ -0,0 +1,155 @@
+
+
+@dataclass
+class RedisUploader(Uploader):
+    upload_config: RedisUploaderConfig
+    connection_config: RedisConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def is_async(self) -> bool:
+        return True
+
+    def precheck(self) -> None:
+        if not self.connection_config.access_config.uri and not self.connection_config.host:
+            raise ValueError("Please pass a hostname either directly or through uri")
+
+        async def check_connection():
+            async with self.connection_config.create_client() as async_client:
+                await async_client.ping()
+
+        try:
+            asyncio.run(check_connection())
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    @requires_dependencies(["redis"], extras="redis")
+    async def run_async(self, path: Path, file_data: FileData, **kwargs: Any) -> None:
+        from redis import exceptions as redis_exceptions
+
+        with path.open("r") as file:
+            elements_dict = json.load(file)
+
+        if elements_dict:
+            logger.info(
+                f"writing {len(elements_dict)} objects to destination asynchronously, "
+                f"db, {self.connection_config.database}, "
+                f"at {self.connection_config.host}",
+            )
+            async with self.connection_config.create_client() as async_client:
+                async with async_client.pipeline(transaction=True) as pipe:
+                    first_element = elements_dict[0]
+                    element_id = first_element["element_id"]
+                    redis_stack = True
+                    try:
+                        # Redis with stack extension supports JSON type
+                        await pipe.json().set(element_id, "$", first_element).execute()
+                    except redis_exceptions.ResponseError as e:
+                        message = str(e)
+                        if "unknown command `JSON.SET`" in message:
+                            # if this error occurs, Redis server doesn't support JSON type,
+                            # so save as string type instead
+                            await pipe.set(element_id, json.dumps(first_element)).execute()
+                            redis_stack = False
+                        else:
+                            raise e
+
+                    for chunk in batch_generator(elements_dict[1:], self.upload_config.batch_size):
+                        for element in chunk:
+                            element_id = element["element_id"]
+                            if redis_stack:
+                                pipe.json().set(element_id, "$", element)
+                            else:
+                                pipe.set(element_id, json.dumps(element))
+                        await pipe.execute()
```

Comment:
At the moment, none of this code is doing any async functions in parallal, meaning it's be slower than simply doing it non-async in the pipeline. We should either update the code to send the commands in parallel using something like `async.gather` or convert it to a normal sync code and let the pipeline run this with a multiprocessing fan out.